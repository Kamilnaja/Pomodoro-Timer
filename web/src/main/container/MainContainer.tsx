import { resetForm, setUserIsLoggedIn, setUserIsLoggedOut } from 'auth/store/actions/authActions';
import React from 'react';
import { connect } from 'react-redux';
import { ModalType as ModalEnum } from 'shared/modal/modalEnum';
import { AuthState } from '../../auth/store/interfaces/authState';
import { CloseModal, OpenModal } from '../../shared/modal/ModalProps';
import { MainComponent } from '../component/MainComponent';
import { MainContainerProps } from './MainContainerProps';
import { MainContainerState } from './MainContainerState';

class MainContainer extends React.Component<MainContainerProps, MainContainerState> {
  constructor(props: MainContainerProps) {
    super(props);
    this.state = {
      openedModal: ModalEnum.NULL,
    };
    this.props.setUserIsLoggedIn();
  }

  handleOpenModal: OpenModal = (modal: ModalEnum) =>
    this.setState({
      openedModal: modal,
    });

  handleCloseModal: CloseModal = () => {
    this.setState({
      openedModal: ModalEnum.NULL,
    });
    this.props.resetForm();
  };

  render = () => (
    <MainComponent
      handleOpenModal={this.handleOpenModal}
      handleCloseModal={() => this.handleCloseModal()}
      openedModal={this.state.openedModal}
      isLoggedIn={this.props.auth.isLoggedIn}
    />
  );
}

const mapDispatchToProps = {
  resetForm,
  setUserIsLoggedIn,
  setUserIsLoggedOut,
};

const mapStateToProps = (state: { auth: AuthState }) => {
  const authState = state.auth;
  return { auth: authState };
};

export default connect(mapStateToProps, mapDispatchToProps)(MainContainer);
