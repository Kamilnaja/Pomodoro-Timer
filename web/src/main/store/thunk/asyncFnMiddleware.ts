export async function fetchStats(dispatch: any, getState: any) {
  const response = await client.get("/fakeApi/todos");
  dispatch({ type: "todos/todosLoaded", payload: response.todos });
}
