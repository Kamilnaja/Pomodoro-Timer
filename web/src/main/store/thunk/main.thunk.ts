import TodayStatistics from "../../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
  savePomodoro,
  savePomodoroError,
  savePomodoroSuccess,
} from "../actions/actions";
// todo - different versions for prod etc
const API_URL = "http://localhost:8080/pomodoros";

function makePostRequest() {
  return fetch(API_URL, {
    method: "POST",
  });
}

export function makeGetRequest() {
  return fetch(API_URL);
}

function handleErrors(response: any): Promise<any> {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response.json();
}

export function savePomodoroThunk() {
  return (dispatch: Function) => {
    dispatch(savePomodoro());

    return makePostRequest()
      .then(handleErrors)
      .then(() => dispatch(savePomodoroSuccess()))
      .catch((error) => dispatch(savePomodoroError(error)));
  };
}

export function getPomodorosThunk() {
  return (dispatch: Function) => {
    dispatch(getTodayStatistics());

    return makeGetRequest()
      .then(handleErrors)
      .then((payload: TodayStatistics) => {
        console.log(payload);
        return dispatch(getTodayStatisticsSuccess(payload));
      })
      .catch((error) => dispatch(getTodayStatisticsError(error)));
  };
}
