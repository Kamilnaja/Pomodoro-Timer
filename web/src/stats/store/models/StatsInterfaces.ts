import { PomodorosDoneInDay } from '../../../../../types/statisticsInterfaces';

export interface StatsState {
  isLoading: boolean;
  error: string;
  results: PomodorosDoneInDay[];
  todayResults: number;
}

export type HandleGetStats = (year?: number, month?: number) => void;
