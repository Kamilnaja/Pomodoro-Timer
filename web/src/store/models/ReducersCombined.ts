export type ReducersCombined = Partial<Record<Reducers, Object>>;

export type Reducers = 'stats' | 'auth' | 'tags' | 'settings' | 'pomodoroCounter';
