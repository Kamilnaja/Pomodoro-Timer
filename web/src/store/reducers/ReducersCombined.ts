export type ReducersCombined = Partial<Record<Reducers, Object>>;

export type Reducers = 'stats' | 'auth' | 'tasksAndNotes' | 'tags' | 'settings' | 'pomodoroCounter';
