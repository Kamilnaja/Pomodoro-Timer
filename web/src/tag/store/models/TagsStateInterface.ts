import { Tag } from '../../../../../types/statsInterfaces';

export interface TagsState {
  isLoading: boolean;
  error: string;
  tags: Tag[];
  currentTag: string;
}
