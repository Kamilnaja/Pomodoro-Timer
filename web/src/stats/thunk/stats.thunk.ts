import { handleErrors } from "../../main/store/thunk/main.thunk";
import { initialConfig } from "../../shared/settings/initialConfig";
import TodayStatistics from "../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
} from "../store/actions/stats.actions";

const API_URL = initialConfig.apiUrl;

const makeGetRequest = () => fetch(API_URL + "/pomodoros_done_today");

export const getPomodorosThunk = () => (dispatch: Function) => {
  dispatch(getTodayStatistics());

  return makeGetRequest()
    .then(handleErrors)
    .then((response) => response.json())
    .then((payload: TodayStatistics) => {
      return dispatch(getTodayStatisticsSuccess(payload));
    })
    .catch((error) => dispatch(getTodayStatisticsError(error)));
};
