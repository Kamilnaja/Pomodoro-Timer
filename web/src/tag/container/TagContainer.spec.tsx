import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createComputedPropertyName } from 'typescript';
import { createAuthState } from '../../auth/testing/auth.testing.data';
import MainContainer from '../../main/container/MainContainer';
import { createPomodoroCounterState } from '../../pomodoroCounter/testing/pomodoroCounter.test.data';
import { createSettingsState } from '../../settings/testing/settings.test.data';
import { createStatsState } from '../../stats/testing/stats.test.data';
import { Reducers, ReducersCombined } from '../../store/models/ReducersCombined';
import { createTags, createTagsState } from '../testing/tags.test.data';

const mockStore = configureStore([thunk]);

describe('TagsContainer', () => {
  let store;
  let component: renderer.ReactTestRenderer;

  const states: ReducersCombined = {
    stats: createStatsState(),
    auth: createAuthState(true),
    pomodoroCounter: createPomodoroCounterState(),
    tags: createTagsState(),
    settings: createSettingsState(),
  };

  beforeEach(() => {
    store = mockStore(states);

    component = renderer.create(
      <Provider store={store}>
        <MainContainer />
      </Provider>,
    );
  });

  it('should match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
