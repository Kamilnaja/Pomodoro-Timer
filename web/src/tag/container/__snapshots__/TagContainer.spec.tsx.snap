// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsContainer should match snapshot 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <nav
      className="navbar navbar-expand navbar-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Tomatodone
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="button--login nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Login
          </a>
          <a
            className="button--register nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Register
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/about"
            href="/about"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </nav>
    <div
      className="align-items-center d-flex flex-column bg-dark jumbotron"
    >
      <div
        className="d-flex justify-content-around w-100"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Pomodoro
        </button>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Short Break
        </button>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Long Break
        </button>
      </div>
      <p
        className="time text-light d-flex align-items-center"
      >
        41:40
      </p>
      <button
        className="btn--pause btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Pause timer
      </button>
      <div
        className="info mt-4"
      >
        <div
          className="text-danger text-center"
        >
          <span>
            Please login to save pomodoros
          </span>
        </div>
      </div>
      <div
        className="error"
      >
        <span>
          Error while trying to synchronize
        </span>
      </div>
    </div>
  </div>
</div>
`;
